MAUI/DataGrid/Column-types.md

```diff
@@ -521,6 +580,227 @@ dataGrid.Columns.Add(dateColumn);
+
+* Record – data object associated with row.
+* Data Context – Binding context of data grid.
+
+In the below code, ItemsSource for ShipCity column returned based on ShipCountry column value using the record and Binding context of data grid passed to GetItemsSource method.
+
+{% tabs %}
+{% highlight xaml %}
+    <ContentPage.Resources>
+        <ResourceDictionary>
+            <local:ItemSourceSelector x:Key="converter" />
+        </ResourceDictionary>
+    </ContentPage.Resources>
+
+    <sfgrid:SfDataGrid x:Name="dataGrid"
+                       ItemsSource="{Binding DealerInformation}"
+                       AllowEditing="True"
+                       AutoGenerateColumnsMode="None"
+                       NavigationMode="Cell"
+                       EditTapAction="OnDoubleTap"
+                       SelectionMode="Single">
+        <sfgrid:SfDataGrid.Columns>
+            <sfgrid:DataGridComboBoxColumn BindingContext="{x:Reference viewModel}"
+                                           ItemsSource="{Binding CountryList}"
+                                           MappingName="ShipCountry"
+                                           LoadUIView="True">
+            </sfgrid:DataGridComboBoxColumn>
+
+            <sfgrid:DataGridComboBoxColumn ItemsSourceSelector="{StaticResource converter}"
+                                           MappingName="ShipCity"
+                                           LoadUIView="True">
+            </sfgrid:DataGridComboBoxColumn>
+        </sfgrid:SfDataGrid.Columns>
+    </sfgrid:SfDataGrid>
+{% endhighlight %}
+
+{% highlight c# %}
+public class ItemSourceSelector : IItemsSourceSelector
+{
+    public IEnumerable GetItemsSource(object record, object dataContext)
+    {
+        if (record == null)
+        {
+            return null;
+        }
+
+        var orderinfo = record as DealerInfo;
+        var countryName = orderinfo.ShipCountry;
+        var viewModel = dataContext as EditingViewModel;
+
+        // Returns ShipCity collection based on ShipCountry.
+        if (viewModel.ShipCities.ContainsKey(countryName))
+        {
+            string[] shipcities = null;
+            viewModel.ShipCities.TryGetValue(countryName, out shipcities);
+            return shipcities.ToList();
+        }
+
+        return null;
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+![DataGrid with ItemSourceSelector comboBox column](Images\column-types\maui-datagrid-comboBox-column-itemsourceselector.png)
+![DataGrid with ItemSourceSelector comboBox column](Images\column-types\maui-datagrid-comboBox-column-itemsourceselector2.png)
+
+### Editing the combo box
+
+The [DataGridComboBoxColumn]() supports both editable and non-editable text box to choose selected items in given data source. Users can select one item from the suggestion list.
+
+[IsEditableMode]() property is used to enable the user input in [DataGridComboBoxColumn](). The default value is `false`.
```

Comment:
The [DataGridComboBoxColumn]() supports both editable and non-editable text boxes for selecting items from a given data source. Users can choose one item from the suggestion list.

The [IsEditableMode]() property is used to enable user input in the [DataGridComboBoxColumn](). Its default value is `false`.

Reply:
![image](https://github.com/syncfusion-content/maui-docs/assets/113881396/56415654-4397-4ff2-9275-93571a301fde)